src/connect.test.js

```diff
@@ -0,0 +1,122 @@
+        state++;
+        return {state};
+      };
+    })());
+    const Wrapped = () => <div/>;
+    const Connected = connect(mapStateToProps)(Wrapped);
+    const helper = {};
+    const state = {};
+    let subscribeListener;
+    const algoliaStore = {
+      getHelper() { return helper; },
+      getState() { return state; },
+      subscribe: fn => { subscribeListener = fn; }
+    };
+    const wrapper = mount(<Connected prop />, {context: {algoliaStore}});
+    subscribeListener();
+    expect(wrapper.find(Wrapped).props()).toEqual({helper, state: 2, prop: true});
+  });
+
+  it('calls store.unsubscribe when component unmount', () => {
+    const unsubscribe = jest.fn();
+    const Connected = connect(() => {})(() => <div/>);
+    const algoliaStore = {getHelper() {}, getState() {}, subscribe() { return unsubscribe; }};
+    const wrapper = mount(<Connected />, {context: {algoliaStore}});
+    expect(unsubscribe).not.toBeCalled();
+    wrapper.unmount();
+    expect(unsubscribe).toBeCalled();
+  });
+
+  it('sets the right displayName on classes', () => {
+    class SuperComponent extends Component {
+      render() {
+        return <div/>;
+      }
+    }
+
+    const algoliaStore = {getHelper() {}, getState() {}, subscribe() {}};
+    const Connected = connect()(SuperComponent);
+    const wrapper = mount(<Connected />, {context: {algoliaStore}});
+    expect(wrapper.name()).toEqual('AlgoliaSearchHelperConnect(SuperComponent)');
+  });
+
+  // this works because babel turns arrows functions to named function expressions
+  it('sets the right displayName on stateless components', () => {
+    const SuperComponent = () => <div/>; // => var SuperComponent = function SuperComponent() {}
+    const algoliaStore = {getHelper() {}, getState() {}, subscribe() {}};
+    const Connected = connect()(SuperComponent);
+    const wrapper = mount(<Connected />, {context: {algoliaStore}});
+    expect(wrapper.name()).toEqual('AlgoliaSearchHelperConnect(SuperComponent)');
+  });
+
+  it('sets the right displayName on React.createClass', () => {
+    const SuperComponent = React.createClass({
+      render() { return <div/>; },
+      displayName: 'SuperComponent'
+    });
+
+    const algoliaStore = {getHelper() {}, getState() {}, subscribe() {}};
+    const Connected = connect()(SuperComponent);
+    const wrapper = mount(<Connected />, {context: {algoliaStore}});
+    expect(wrapper.name()).toEqual('AlgoliaSearchHelperConnect(SuperComponent)');
+  });
+
+  it('sets a default displayName when not able to find one', () => {
+    const SuperComponent = React.createClass({
+      render() { return <div/>; },
+      displayName: undefined // latest babel
+    });
+
+    const algoliaStore = {getHelper() {}, getState() {}, subscribe() {}};
+    const Connected = connect()(SuperComponent);
+    const wrapper = mount(<Connected />, {context: {algoliaStore}});
+    expect(wrapper.name()).toEqual('AlgoliaSearchHelperConnect(UnknownComponent)');
+  });
+});
```

Comment:
An important feature of connect + mapStateToProps is that we decide whether to update the underlying component based on a shallow comparison of props before and after the store state update. So if a component connects to `searchParameters.page` explicitly, it will only ever get updated when that parameter changes (and when its own props change, of course).